src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -0,0 +1,399 @@
+     */
+    void setRainTime(int time);
+
+    /**
+     * Gets whether this world is currently experiencing a lightning storm.
+     * 
+     * @return Is thundering
+     */
+    boolean isThundering();
+
+    /**
+     * Sets whether this world is currently experiencing a lightning storm.
+     * 
+     * @param state Is thundering
+     */
+    void setThundering(boolean state);
+
+    /**
+     * Gets the number of ticks until the {@link #isThundering()} state is
+     * toggled to a new random value.
+     * 
+     * @return The time until the thundering state changes
+     */
+    int getThunderTime();
+
+    /**
+     * Sets the number of ticks until the {@link #isThundering()} state is
+     * toggled to a new random value.
+     * 
+     * @param time The time until the thundering state changes
+     */
+    void setThunderTime(int time);
+
+    /**
+     * Gets the default {@link GameMode} of this world.
+     * 
+     * @return The game mode
+     */
+    GameMode getGameMode();
+
+    /**
+     * Sets the default {@link GameMode} of this world.
+     * 
+     * @param gamemode The game mode
+     */
+    void setGameMode(GameMode gamemode);
+
+    /**
+     * Gets whether this world will generate map features such as villages and
+     * strongholds.
+     * 
+     * @return Whether map features enabled
+     */
+    boolean usesMapFeatures();
+
+    /**
+     * Sets whether this world will generate map features such as villages and
+     * strongholds.
+     * 
+     * @param state Whether map features enabled
+     */
+    void setMapFeaturesEnabled(boolean state);
+
+    /**
+     * Gets whether this world is set to hardcore mode.
+     * 
+     * @return Is hardcore
+     */
+    boolean isHardcore();
+
+    /**
+     * Sets whether this world is set to hardcore mode.
+     * 
+     * @param state Is hardcore
+     */
+    void setHardcore(boolean state);
+
+    /**
+     * Gets whether commands are allowed within this world. May not be respected
+     * when not in single player.
+     * 
+     * @return Whether commands are allowed
+     */
+    boolean areCommandsAllowed();
+
+    /**
+     * Sets whether commands are allowed within this world. May not be respected
+     * when not in single player.
+     * 
+     * @param state Whether commands are allowed
+     */
+    void areCommandsAllowed(boolean state);
+
+    /**
+     * Gets whether this world has been initialized.
+     * 
+     * @return Is initialized
+     */
+    boolean isInitialized();
+
+    /**
+     * Gets the difficulty of this world.
+     * 
+     * @return The difficulty
+     */
+    Difficulty getDifficulty();
+
+    /**
+     * Sets the difficulty of this world.
+     * 
+     * @param difficulty The difficulty
+     */
+    void setDifficulty(Difficulty difficulty);
+
+    /**
+     * Gets the {@link WorldBorder} of this world.
+     * 
+     * @return The world border
+     */
+    WorldBorder getWorldBorder();
```

Comment:
WorldBorder is mutable, why are you wanting to set it? Setting the world border is not something you've ever been able to do to a world, nor should you.